src/environment.js

```diff
@@ -120,7 +119,11 @@ var Environment = Obj.extend({
             throw new Error('template names must be a string: ' + name);
         }
 
-        var tmpl = this.cache[name];
+        for (var i = 0; i < this.loaders.length; i++) {
+            var _name = parentName? this.loaders[i].resolve(parentName, name) : name;
+            tmpl = this.loaders[i].cache[_name];
```

Comment:
That's true. I'm not sure I love that we are going down a route of allowing almost anything to be possible (sometimes constraints make it easier to reason about any nunjucks project), but I think this project generally is evolving that way, so I'm fine with that. At this point I'm pretty much open to whatever you all need.